hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/entity/EntityRecordItemListener.java

```diff
@@ -236,6 +245,8 @@ public void onItem(RecordItem recordItem) throws ImporterException {
             insertAutomaticTokenAssociations(recordItem);
         }
 
+        insertContractResult(recordItem, transactionHandler.getContractResult(transaction, recordItem));
```

Comment:
Would suggest adding a `TransactionHandler::onContractResult` that can be passed to the `ContractResultService` to customize the created `ContractResult` for different transaction types.

```suggestion
        contractResultService.process(recordItem, transactionHandler::onContractResult);
```

Reply:
Updated logic to pass transactionHandler through and use in Service